ui/selectors/selectors.js

```diff
@@ -209,6 +209,17 @@ export function isHardwareWallet(state) {
   return Boolean(keyring?.type?.includes('Hardware'));
 }
 
+/**
+ * Checks if the current account is a Snap account.
+ *
+ * @param {object} state - The state object.
+ * @returns {boolean}
+ */
+export function isAccountSnap(state) {
+  const keyring = getCurrentKeyring(state);
+  return Boolean(keyring?.type?.includes('Snap'));
+}
```

Comment:
Can we use a constant value here, something like,

```suggestion
export function isAccountSnap(state) {
  const keyring = getCurrentKeyring(state);
  return keyring?.type === KeyringType.snap;
}
```

Also, I see we have the selector `getAccountType`, lets make sure we're not duplicating code unnecessarily